src/server/components/annual-review/controller.ts

```diff
@@ -9,14 +9,23 @@ import { getDetailsViewModel } from "server/components/dashboard/listsItems/getV
 import { getCSRFToken } from "server/components/cookies/helpers";
 import { HttpException } from "server/middlewares/error-handlers";
 import { prisma } from "server/models/db/prisma-client";
+import { logger } from "server/services/logger";
 
 import type { ListItemGetObject, List } from "server/models/types";
 import { EVENTS } from "server/models/listItem/listItemEvent";
+import { initialiseFormRunnerSession } from "server/components/formRunner/helpers";
 
 export async function confirmGetController(req: Request, res: Response, next: NextFunction): Promise<void> {
   try {
     const { listItemRef } = req.params;
-    const listItem = (await findListItemByReference(listItemRef)) as ListItemGetObject;
+    const result = await findListItemByReference(listItemRef);
+
+    if (!result) {
+      return next(new HttpException(404, "404", "The list item cannot be found"));
+    }
+
+    const listItem = result as unknown as ListItemGetObject;
```

Comment:
what's the impact of this line with the cast to unknown in the middle?

Reply:
So we're not force casting `ListItemGetObject` everywhere. prev code had force casts `const foo = bar as ListItemGetObject` in a few places, which caused further bugs/issues. It's related to this issue found here https://github.com/UKForeignOffice/lists/pull/498#discussion_r1047460086


1. findByReference was not returning address (`select|include: { address: true }`)
2. the prisma response from the above was being cast to `ListItemGetObject` which includes the `address` object (even though it did not have it)

I managed to remove most of the casts, so letting tsc handle and infer types by itself. This was the last one. resolving it probably would've caused too many code changes and I'm keen to get this through. 

tsc won't let us cast `result as ListItemGetObject` here, I forget why, some issue with non-overlapping types. I'm inclined to think there are a bunch of null, undefined, or unions which were causing issues (probably from other force casts 🙂). 

`const listItem = result as unknown as ListItemGetObject;`
"I don't know if this is ListItemGetObject, or null, or undefined, or whatever else, but just cast it to ListItemGetObject".